test/jdk/jdk/nio/zipfs/TestLocOffsetFromZip64EF.java

```diff
@@ -143,88 +140,122 @@ public void walkZipFileTest() throws IOException {
+     * - The sizes and offset fields values of the CEN are set to 0xFFFFFFFF (Zip64 magic values)
+     *
      */
-    private static void createFiles() throws IOException {
-        try (RandomAccessFile file = new RandomAccessFile(LARGE_FILE_NAME, "rw")
-        ) {
-            System.out.printf("Creating %s%n", LARGE_FILE_NAME);
-            file.setLength(LARGE_FILE_SIZE);
-            System.out.printf("Creating %s%n", SMALL_FILE_NAME);
-            Files.writeString(Path.of(SMALL_FILE_NAME), "Hello");
-        }
-    }
+    public void createZipWithZip64Ext() throws IOException {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try (ZipOutputStream zo = new ZipOutputStream(out)) {
 
-    /**
-     * Utility method to execute a ProcessBuilder command
-     * @param pb ProcessBuilder to execute
-     * @return The Result(s) from the ProcessBuilder execution
-     */
-    private static Result run(ProcessBuilder pb) {
-        Process p;
-        System.out.printf("Running: %s%n", pb.command());
-        try {
-            p = pb.start();
-        } catch (IOException e) {
-            throw new RuntimeException(
-                    format("Couldn't start process '%s'", pb.command()), e);
-        }
+            ZipEntry e = new ZipEntry("entry");
+            // Add an entry, make it STORED and empty to simplify parsing
+            e.setMethod(ZipEntry.STORED);
+            e.setSize(0);
+            e.setCrc(0);
+            zo.putNextEntry(e);
 
-        String output;
-        try {
-            output = toString(p.getInputStream(), p.getErrorStream());
-        } catch (IOException e) {
-            throw new RuntimeException(
-                    format("Couldn't read process output '%s'", pb.command()), e);
-        }
+            // Add an additional entry as a sanity check that we can navigate past the first
+            ZipEntry e2 = new ZipEntry("entry2");
+            e2.setMethod(ZipEntry.STORED);
+            e2.setSize(0);
+            e2.setCrc(0);
+            zo.putNextEntry(e2);
+
+            zo.closeEntry(); // At this point, all LOC headers are written.
 
-        try {
-            p.waitFor();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(
-                    format("Process hasn't finished '%s'", pb.command()), e);
+            // We want the first CEN entry to have two extra fields:
+            // 1: A 'Info-Zip extended timestamp' extra field, generated by ZipOutputStream
+            //       when the following date fields are set:
+            e.setLastModifiedTime(FileTime.from(Instant.now()));
+            e.setLastAccessTime(FileTime.from(Instant.now()));
+
+            // 2: An opaque extra field, right-sized for a Zip64 extended field,
+            //    to be updated below
+            byte[] zip64 = makeOpaqueExtraField();
+            e.setExtra(zip64);
+
+            zo.finish(); // Write out CEN and END records
```

Comment:
I might add one more entry after this one, perhaps with a compression method of DEFLATED